apps/todolist-mariadb-go/todolist.go

```diff
@@ -33,13 +33,52 @@ import (
 
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/gorilla/mux"
-	"github.com/jinzhu/gorm"
 	_ "github.com/jinzhu/gorm/dialects/mysql"
 	"github.com/rs/cors"
 	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
 )
 
-var db, _ = gorm.Open("mysql", "changeme:changeme@(mysql:3306)/todolist?charset=utf8&parseTime=True")
+var db *gorm.DB
+
+func connectToDB() {
+	remote, err := connectToMariaDBRemote()
+	if err != nil {
+		local, _ := connectToMariaDBLocal()
+		db = local
+	} else {
+		db = remote
+	}
+}
+
+// connect to mariadb at 127.0.0.1
```

Comment:
In python you are using localhost here 127.0.0.1, please use some consistent resolve or IP.

Reply:
You could also simplify this by:

```go
func connectToDB() {
	if db, _ = connectToMariaDBRemote(); db == nil {
		db, _ = connectToMariaDBLocal()
	}
}
```